code/game/gamemodes/vampire/vampire.dm

```diff
@@ -318,43 +323,58 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 
 	// TIER 1
 	if(vamp.bloodtotal >= 100)
-		if(!(VAMP_VISION in vamp.powers))
+		if(!(vamp.powers |= VAMP_VISION))
 			vamp.powers.Add(VAMP_VISION)
-		if(!(VAMP_SHAPE in vamp.powers))
+		if(!(vamp.powers |= VAMP_SHAPE))
 			vamp.powers.Add(VAMP_SHAPE)
 
 	// TIER 2
 	if(vamp.bloodtotal >= 150)
-		if(!(VAMP_CLOAK in vamp.powers))
+		if(!(vamp.powers |= VAMP_CLOAK))
 			vamp.powers.Add(VAMP_CLOAK)
-		if(!(VAMP_DISEASE in vamp.powers))
+		if(!(vamp.powers |= VAMP_DISEASE))
 			vamp.powers.Add(VAMP_DISEASE)
```

Comment:
the check isn't necessary if you're using ```|=```, the ```|=``` itself means "add it to the list if it isn't already there".
So this:
```
if(!(vamp.powers |= VAMP_DISEASE))
 			vamp.powers.Add(VAMP_DISEASE)
```
Should look like this:
```
vamp.powers |= VAMP_DISEASE
```